spec.md

```diff
@@ -4292,7 +4294,15 @@ decl_layer =
   dedent ;
 
 decl_formal =
-  "formal" , id , "of" , id , "," , "bound" , "=" , int;
+  "formal" , id , "of" , id , ":" , [ info ] , newline , indent ,
+    { id , "=" , decl_formal_param , newline } ,
+  dedent ;
+decl_formal_param =
+    int
+  | string_dq
+  | string_sq
+  | "[" , [ decl_formal_param , { "," , decl_formal_param } ] , "]"
+  | "{" , [ id , "=" , decl_formal_param , { "," , id , "=" , decl_formal_param } ] , "}"
```

Comment:
Got it. So, numbers, strings, and lists or dictionaries of the same types. Seems reasonable and flexible.